docs/lib/regex.rst

```diff
@@ -0,0 +1,57 @@
+.. _regex:
+
+Scala Native Regexes
+=====================
+
+Scala Native implements java.util.regex with `Google's RE2 library <https://github.com/google/re2>`_. There is some differences with the original API:
+
+Some expressions are not supported:
+ * Character Classes
+   * unions: [a-d[m-p]]
+   * intersections: [a-z&&[^aeiou]]
+ * predefined character classes: \h, \H, \v, \V
+ * java character function classes
+   * \p{javaLowerCase}
+   * \p{javaUpperCase}
+   * \p{javaWhitespace}
+   * \p{javaMirrored}
+ * boundary matchers: \G, \Z, \R
+ * possessive quantifiers
+   * X?+
+   * X*+
+   * X++
+   * X{n}+
+   * X{n,}+
+   * X{n,m}+
+ * lookaheads
+   * (?=X)
+   * (?!X)
+   * (?<=X)
+   * (?<!X)
+   * (?>X)
+
+Some expressions have an alternative syntax:
+
+========== =============
+Java       RE2
+========== =============
+Java       RE2
```

Comment:
Converting via find-and-replace is really dangerous. RegExes are its own language and we'll probably need to parse it to do something like this in a sound manner.